metricbeat-yml-script.py

```diff
@@ -40,44 +41,37 @@ def _is_open():
 
 
 def _add_modules():
-    if _custom_modules():
-        logger.debug("Using custom modules")
-        return
     try:
         modules = [m.strip() for m in os.environ["LOGZIO_MODULES"].split(",")]
     except KeyError:
         logger.error("Required at least one module")
         raise
-
     _enable_modules(modules)
 
 
-def _custom_modules():
-    return len(glob.glob("{}/*.yml".format(MODULES_DIR)))
-
-
 def _enable_modules(modules):
     yaml = YAML()
-    supported_modules = dict((name, setup) for name, setup in setups)
+    yaml.preserve_quotes = True
     for module in modules:
-        if module in supported_modules:
-            conf = supported_modules[module]()
-            with open("{0}/{1}{2}".format(MODULES_DIR, module, ".yml"), "w+") as conf_yaml:
-                logger.debug("Adding the following conf: {}".format(conf))
-                yaml.dump(conf, conf_yaml)
-        else:
+        if module not in SUPPORTED_MODULES:
             logger.error("Unsupported module: {}".format(module))
             raise RuntimeError
+        with open("modules/{}.yml".format(module)) as module_file:
+            module_yaml = yaml.load(module_file)
+            module_yaml[0]["enabled"] = True
+        with open("modules/{}.yml".format(module), "w+") as module_file:
+            yaml.dump(module_yaml, module_file)
```

Comment:
Why twice open?

Reply:
once for read and once for write

Reply:
Can't we do it in one `with` statement using `r+`?